public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx

```diff
@@ -87,8 +89,7 @@ export function DashboardSettings({ dashboard, editview, pageNav, sectionNav }:
   );
 }
 
-function getSettingsPages(dashboard: DashboardModel) {
-  const { t } = useTranslate();
+function getSettingsPages(dashboard: DashboardModel, t: TFunction) {
```

Comment:
nit: If we want to avoid importing from `internal`, could do `ReturnType<typeof useTranslate>['t']` here as well, or we just export the type of the `t` function from the main export anyway

Reply:
Good catch 🙏 , I'm thinking we should move the `TFunction` type from `internal` to the `public` export. But I'll do that in a follow up PR 👍

Reply:
Opened https://github.com/grafana/grafana/issues/105920 and I have a PR ready soon 👍